### ISSUE TITLE
[feature request] Super+Right Click tap to open cosmic-comp context menu

### ISSUE THREAD

**Issue by alphabitserial** [pop-os/cosmic-comp#909]
Right clicking the title bar in a libcosmic app exposes a lot of useful things: creating window stacks, taking screenshots, minimize/maximize, moving windows around workspaces, etc.  However, there is currently no way to bring up this menu in, for example, a GTK app which uses client-side decorations!

Right now, holding Super, then holding right click allows for resizing a window. It would be great if holding super and *tapping* right click would open that context menu in any application. As a bonus, it also allows for opening it in COSMIC applications without having to first aim the pointer at the titlebar.

**Comment by Drakulix** [comment]
>  However, there is currently no way to bring up this menu in, for example, a GTK app which uses client-side decorations!

This isn't true, given that GTK apps send the required wayland-request on their title bar to make cosmic-comp open the menu.

However there are windows completely devote of decorations and some less known or cross-platform toolkits, which don't do this.

@maria-komarova I have thought about how we could implement this without loosing resize, given this is an issue for e.g. Firefox Picture-and-picture windows (something many people probably want to be sticky, which currently can only be done through the shortcut).

I think this is a decent proposal, though I also thought about using Super+Middle-Click, however not every pointer device does have that button. Any opinion from a UX perspective?

**Comment by maria-komarova** [comment]
I'd try to avoid using middle click for the reason you mention - not every pointer device has the button. So using right click on press and release seems like a better option. I might be missing some potential concerns but I can't yet see them.

**Comment by alphabitserial** [comment]
> This isn't true, given that GTK apps send the required wayland-request on their title bar to make cosmic-comp open the menu.
My mistake, you're right. Firefox was the case I noticed it not working in, but it makes sense that they'd be doing their own thing.

**Comment by Drakulix** [comment]
> > This isn't true, given that GTK apps send the required wayland-request on their title bar to make cosmic-comp open the menu.
> 
> My mistake, you're right. Firefox was the case I noticed it not working in, but it makes sense that they'd be doing their own thing.

Note for firefox specifically: You can do a second right-click and it will work.

But we will implement this (eventually).

**Comment by hojjatabdollahi** [comment]
Is there a shortcut for "Always on top"? Sticky will keep the window even when you change workspaces. 
I'm curious if there is a "Always on top" action and if it could be assigned to a shortcut (since probably exposing it in the context menu would cause confusion with the sticky window feature).

**Comment by onlyreportingissues** [comment]
> Is there a shortcut for "Always on top"? Sticky will keep the window even when you change workspaces. I'm curious if there is a "Always on top" action and if it could be assigned to a shortcut (since probably exposing it in the context menu would cause confusion with the sticky window feature).

Would be cool to have such an option. Maybe it could also work then the way it does on GNOME: https://github.com/pop-os/cosmic-comp/issues/534

**Comment by Drakulix** [comment]
> I'm curious if there is a "Always on top" action

No. This concept is not present in cosmic-comp and we don't have an action for it.
